File: app.main/ticket_app.py
import tkinter as tk
from tkinter import ttk, messagebox
import json

import sys

# Add the path of your script to sys.path
module_path = '../web_scraper'
if module_path not in sys.path:
    sys.path.append(module_path)
print(module_path)

# Import the script as a module
import get_events

# Call the function from the script
get_events.main()


class TicketApp(tk.Tk):
    def __init__(self, json_path):
        super().__init__()

        self.title("Ticket Purchase App")
        self.geometry("300x200")

        # Read dates from JSON at a given path
        self.ticket_types = self.read_ticket_types(json_path)
        self.ticket_quantity = tk.IntVar(value=1)

        # Widgets
        self.create_widgets()

    def read_ticket_types(self, filepath):
        """Reads dates from the 'date' key in a JSON file and returns a list of dates."""
        types = []
        try:
            with open(filepath, 'r', encoding='utf-8') as jsonfile:
                data = json.load(jsonfile)
                types = [item['date'] for item in data if 'date' in item]
        except FileNotFoundError:
            messagebox.showerror("File Not Found", f"The file at {filepath} was not found.")
        except json.JSONDecodeError:
            messagebox.showerror("Error", "The JSON file is malformed.")
        except Exception as e:
            messagebox.showerror("Error", str(e))
        return types

    def create_widgets(self):
        # Ticket Type Dropdown
        ttk.Label(self, text="Select Date:").pack(pady=10)
        self.ticket_type_combo = ttk.Combobox(self, values=self.ticket_types, state="readonly")
        self.ticket_type_combo.pack()
        self.ticket_type_combo.current(0)

        # Ticket Quantity Entry
        ttk.Label(self, text="Enter Quantity:").pack(pady=10)
        self.quantity_entry = ttk.Entry(self, textvariable=self.ticket_quantity)
        self.quantity_entry.pack()

        # Purchase Button
        ttk.Button(self, text="Purchase", command=self.purchase_tickets).pack(pady=20)

    def purchase_tickets(self):
        ticket_type = self.ticket_type_combo.get()
        quantity = self.ticket_quantity.get()
        message = f"You have purchased {quantity} tickets for the date {ticket_type}."
        messagebox.showinfo("Purchase Confirmation", message)

# Run the application with a path to the JSON
if __name__ == "__main__":
    json_file_path = "../web_scraper/events.json"  # Adjust this path as needed
    app = TicketApp(json_file_path)
    app.mainloop()

File: app.main/ticket_app.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"log"
	"os"
	"path/filepath"
	"strconv"
	"strings"

	"github.com/andlabs/ui"
)

type TicketType struct {
	Date string
}

func main() {
	// Call the function from the script
	getEvents()

	// Run the application with a path to the JSON
	jsonFilePath := "../web_scraper/events.json"
	app := NewTicketApp(jsonFilePath)
	app.Run()
}

func getEvents() {
	modulePath := "../web_scraper"
	if !strings.Contains(os.Getenv("GOPATH"), modulePath) {
		os.Setenv("GOPATH", filepath.Join(os.Getenv("GOPATH"), modulePath))
	}
	// Call the function from the script
	// get_events.main()
}

type TicketApp struct {
	window         *ui.Window
	ticketTypes    []TicketType
	ticketQuantity int
}

func NewTicketApp(jsonPath string) *TicketApp {
	app := &TicketApp{}

	// Read dates from JSON at a given path
	app.ticketTypes = app.readTicketTypes(jsonPath)

	// Create widgets
	app.createWidgets()

	return app
}

func (app *TicketApp) readTicketTypes(filepath string) []TicketType {
	var types []TicketType
	data, err := ioutil.ReadFile(filepath)
	if err != nil {
		log.Printf("Error reading file: %v", err)
		return types
	}

	var jsonData []map[string]interface{}
	err = json.Unmarshal(data, &jsonData)
	if err != nil {
		log.Printf("Error decoding JSON: %v", err)
		return types
	}

	for _, item := range jsonData {
		if date, ok := item["date"].(string); ok {
			types = append(types, TicketType{Date: date})
		}
	}

	return types
}

func (app *TicketApp) createWidgets() {
	// Create the main window
	app.window = ui.NewWindow("Ticket Purchase App", 300, 200, false)
	app.window.SetMargined(true)

	// Create the ticket type dropdown
	ticketTypeBox := ui.NewVerticalBox()
	ticketTypeLabel := ui.NewLabel("Select Date:")
	ticketTypeBox.Append(ticketTypeLabel, false)

	ticketTypeCombo := ui.NewCombobox()
	for _, ticketType := range app.ticketTypes {
		ticketTypeCombo.Append(ticketType.Date)
	}
	ticketTypeBox.Append(ticketTypeCombo, false)

	// Create the ticket quantity entry
	quantityBox := ui.NewVerticalBox()
	quantityLabel := ui.NewLabel("Enter Quantity:")
	quantityBox.Append(quantityLabel, false)

	quantityEntry := ui.NewEntry()
	quantityBox.Append(quantityEntry, false)

	// Create the purchase button
	purchaseButton := ui.NewButton("Purchase")
	purchaseButton.OnClicked(func(*ui.Button) {
		selectedType := ticketTypeCombo.Selected()
		quantity, _ := strconv.Atoi(quantityEntry.Text())
		message := fmt.Sprintf("You have purchased %d tickets for the date %s.", quantity, app.ticketTypes[selectedType].Date)
		ui.MsgBox(app.window, "Purchase Confirmation", message)
	})

	// Add the widgets to the main window
	mainBox := ui.NewVerticalBox()
	mainBox.Append(ticketTypeBox, false)
	mainBox.Append(quantityBox, false)
	mainBox.Append(purchaseButton, false)
	app.window.SetChild(mainBox)
}

func (app *TicketApp) Run() {
	ui.Main(func() {
		app.window.Show()
	})
}
